phases/10-llms-from-scratch/01-tokenizers/code/main.py

```
@@ -0,0 +1,227 @@
+        total_tokens += len(encoded)
+        total_words += len(text.split())
+        for t in encoded:
+            token_usage[t] += 1
+
+    avg_tokens_per_word = total_tokens / total_words if total_words > 0 else 0
+
+    print(f"Vocabulary size: {tokenizer.vocab_size()}")
+    print(f"Avg tokens per word: {avg_tokens_per_word:.2f}")
+    print(f"Total unique tokens used: {len(token_usage)}")
+
+    print(f"\nTop 10 most used tokens:")
```

Ruff (0.15.7)

[error] f-string without any placeholders

Remove extraneous `f` prefix

(F541)